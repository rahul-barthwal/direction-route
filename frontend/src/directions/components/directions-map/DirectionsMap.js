import React, { Component } from 'react';

import { mapClient } from '../../../common/map-client/map-client';

import './DirectionsMap.scss';

/**
 * @name DirectionsMap
 * @type {Component}
 * @extends Component React component
 * @description This component display the map and contains the functionality to render the routes
 */
class DirectionsMap extends Component {
    // map container to save the ref to render the map
    mapContainer;
    // containe the map reference after rendering the map
    map;
    // google map api reference
    google;

    /**
     * @name initMap
     * @description Initialize the map
     */
    initMap = async () => {
        this.google = await mapClient;

        this.map = new this.google.maps.Map(this.mapContainer, {
            zoom: 11,
            center: { lat: 22.372081, lng: 114.107877 }
        });
    };

    drawDirections = ({ path }) => {
        const dirCoordinates = path.map(([lat, lng]) => ({
            lat: Number(lat),
            lng: Number(lng)
        }));

        const dirPath = new this.google.maps.Polyline({
            path: dirCoordinates,
            geodesic: true,
            strokeColor: '#4285f4',
            strokeOpacity: 1.0,
            strokeWeight: 4
        });

        dirPath.setMap(this.map);
    };

    componentDidMount() {
        this.initMap();
    }

    componentWillReceiveProps(props) {
        const { directions } = props;
        if (directions) {
            this.drawDirections(directions);
        }
    }

    render() {
        return (
            <div className="map-container">
                <div ref={el => (this.mapContainer = el)} />
            </div>
        );
    }
}

export default DirectionsMap;
