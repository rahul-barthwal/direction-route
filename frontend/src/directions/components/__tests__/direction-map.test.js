import React from 'react';
import renderer from 'react-test-renderer';

import DirectionsMap from '../directions-map/DirectionsMap';

describe('Direction Form Test', () => {
    it('render correctly', () => {
        const tree = renderer.create(<DirectionsMap />).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
