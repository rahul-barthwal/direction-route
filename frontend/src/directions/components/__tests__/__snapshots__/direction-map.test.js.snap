// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Direction Form Test render correctly 1`] = `
<div
  className="map-container"
>
  <div />
</div>
`;
